packages/components/src/Autocomplete/useRepositionMenu.ts

```diff
@@ -1,31 +1,57 @@
+import { autoUpdate, flip, offset, useFloating } from "@floating-ui/react";
 import { useSafeLayoutEffect } from "@jobber/hooks/useSafeLayoutEffect";
-import { useState } from "react";
-import { usePopper } from "react-popper";
+import { useCallback } from "react";
 import { MenuProps } from "./Autocomplete.types";
 
-export interface UseRepositionMenu extends ReturnType<typeof usePopper> {
+export interface UseRepositionMenu {
   readonly menuRef: HTMLElement | null;
   readonly setMenuRef: (ref: HTMLElement | null) => void;
   readonly targetWidth: number | undefined;
+  readonly styles: {
+    float: React.CSSProperties;
+  };
 }
 
 export function useRepositionMenu(
   attachTo: MenuProps["attachTo"],
-  visible = false,
-) {
-  const [menuRef, setMenuRef] = useState<HTMLElement | null>(null);
-  const popper = usePopper(attachTo.current, menuRef, {
-    modifiers: [
-      { name: "offset", options: { offset: [0, 8] } },
-      { name: "flip", options: { fallbackPlacements: ["top"] } },
-    ],
+  visible: boolean,
+  cssManagedVisibility: boolean,
+): UseRepositionMenu {
+  const { refs, floatingStyles, update } = useFloating({
+    placement: "bottom",
+    middleware: [offset(8), flip({ fallbackPlacements: ["top"] })],
+    elements: {
+      reference: attachTo,
+    },
+    ...(!cssManagedVisibility
+      ? {
+          whileElementsMounted: autoUpdate,
+        }
+      : {}),
   });
 
+  const conditionalUpdate = useCallback(() => {
+    if (cssManagedVisibility && visible && attachTo && refs.floating.current) {
+      const cleanup = autoUpdate(attachTo, refs.floating.current, update);
+
+      return cleanup;
+    }
+
+    return undefined;
+  }, [update, cssManagedVisibility, visible, attachTo, refs.floating.current]);
```

Comment:
I really don't like this but I don't see another way

for `DefaultMenu` it's not rendered when `inputFocused` is false, so we can simply use `whileElementsMounted: autoUpdate,`

for `CustomMenu` it's being hidden with CSS, so we should explicitly NOT use the `whileElementsMounted` because that is going to be constantly firing and calculating when it doesn't need to.

so that means we have to A) avoid using that and B) update manually

the opposite of that is also true, so we have this pretty gross setup :'(

Reply:
@jdeichert definitely looking for feedback on this topic. I am less than happy with this solution.